### PR TITLE
WIP: make Harmony and cylinder device public (first steps)

### PR DESCRIPTION
**Pull request by che85**
fixes https://github.com/JolleyLabCHOP/SlicerHeartPrivate/issues/80
fixes partially https://github.com/JolleyLabCHOP/SlicerHeartPrivate/issues/90

@lassoan run this code on Slicer startup to test it.
```
from CardiacDeviceSimulatorUtils.devices import CylinderDevice
from CardiacDeviceSimulator import CardiacDeviceSimulatorWidget
CardiacDeviceSimulatorWidget.registerDevice(CylinderDevice)
```

After this, open the CardiacDeviceSimulator

### CODE REVIEW

**Comment by che85** [comment]
Note: This is the initial version. I could definitely generalize it more and use device-specific logic/parameters instead of creating separate widgets. But that's just a thought and might not even worth it.

**Comment by che85** [comment]
@lassoan Please take another look. I still have a class `CardiacDeviceWidget `for generating the device widget, but devices itself are UI independently implemented

File: CardiacDeviceSimulator/CardiacDeviceSimulator.py [hunk 1]
```diff
@@ -647,7 +649,7 @@ def setParameterNode(self, parameterNode):
       n = slicer.mrmlScene.CreateNodeByClass("vtkMRMLProceduralColorNode")
       n.UnRegister(None)
       n.SetName(slicer.mrmlScene.GenerateUniqueName("Radial compression (mm)"))
-      n.SetAttribute("Category", "CardiacDeviceSimulator")
+      n.SetAttribute("Category", "CardiacModelGenerator")
```
Comment (che85) [inline CardiacDeviceSimulator/CardiacDeviceSimulator.py]: I am not sure if it should be the same category as it was in the past

File: CardiacDeviceSimulator/CardiacDeviceSimulatorUtils/devices.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      'proximalStraightLengthMm': self.harmonyTcpvProximalStraightLengthSliderWidget.value
+    })
+
+    if self._updateCB:
+      self._updateCB()
+
+  def getProfilePoints(self, params, segment=None, openSegment=True):
+    import math
+
+    #TODO: the following variable is not used at all
```
Comment (che85) [inline CardiacDeviceSimulator/CardiacDeviceSimulatorUtils/devices.py]: @lassoan Is there any reason, why lengthMm was assigned but never used?

Comment (lassoan) [reply CardiacDeviceSimulator/CardiacDeviceSimulatorUtils/devices.py]: Probably was used at some point but not anymore. You can remove it.

File: CardiacDeviceSimulator/CardiacDeviceSimulator.py (hunk 1 above)
Comment (lassoan) [reply CardiacDeviceSimulator/CardiacDeviceSimulator.py]: You've done it right. We can set any name but it is more clear what it is for if the category name matches module name.

Comment (che85) [reply CardiacDeviceSimulator/CardiacDeviceSimulator.py]: Does that affect any of the already created models from the "old" module?

Comment (lassoan) [reply CardiacDeviceSimulator/CardiacDeviceSimulator.py]: It does not affect any old scenes. Category only defines how it shows up in the color node tree.

**Comment by lassoan** [comment]
Thank you, GUI is sufficiently separated from logic now.

**Comment by che85** [comment]
Any other changes that you would suggest? 

One thing I absolutely don't like is that the model gets continuously updated while dragging a slider. Not sure if it makes sense to only update it once mouse button is released or even have a mode "interactive" that can be disabled so that it only updates once slider has been released.

**Comment by lassoan** [comment]
> One thing I absolutely don't like is that the model gets continuously updated while dragging a slider.

You can disable tracking for the slider http://www.commontk.org/docs/html/classctkSliderWidget.html#a401590afb5e1de25506de773b1d5aa7d

I had a look at the go and did a quick testing and it was all good. If you are done with it then you can rebase&merge it.

**Comment by che85** [comment]
@lassoan I don't have the rights to merge this branch

**Comment by lassoan** [comment]
Thank you, I've merged it and invited you to the repository.

**Pull request merged**